### PR TITLE
Add support for bzlmod

### PR DESCRIPTION
**Pull request by jwnrt**
## Summary

This PR does some general cleaning to prepare for and then enable bzlmod support.

`WORKSPACE` support has been removed to reduce maintenance burden and complexity.

Best reviewed by commit.

## Details

The full changes are:

* CI workflow added to build the firmware in PRs.
* `bazel-release` dependency removed, GitHub CLI used directly.
* `crt` dependency swapped for `toolchains_arm_gnu` which supports bzlmod.
* `BUILD` files cleaned up with some indirection and hacks removed.
* `WORKSPACE` files and infrastructure removed.

## Testing

Tested the deployment in my fork here: https://github.com/jwnrt/hyperdebug-firmware/actions/runs/9853140349

Created this release: https://github.com/jwnrt/hyperdebug-firmware/releases/tag/20240709_01

I've tested the release on our FPGA board by running a few tests that go through HyperDebug.

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (pamaury) [inline .github/workflows/ci.yaml:20]: where is the `apt-requirements.txt` sourced from? I can't see it in the repository

File: .github/workflows/create_release.yml
Comment (pamaury) [inline .github/workflows/create_release.yml:23]: I don't necessarily have a strong opinion but I can see an advantage to using bazel-release in theory: the release process is part of the build and can be done locally easily. Github/Azure flows are always notorioulsy bad for local reproducibility as they get more complicated.

File: .github/workflows/ci.yaml
Comment (jwnrt) [reply .github/workflows/ci.yaml:20]: It's in `third_party/chromium/apt-requirements.txt` but contains only `gawk` I copied this from the other workflow but might be better to inline `apt install gawk` here.

File: .github/workflows/create_release.yml
Comment (jwnrt) [reply .github/workflows/create_release.yml:23]: That makes sense.

I have this PR for adding bzlmod support to bazel-release: https://github.com/lowRISC/bazel-release/pull/5

If that's merged, we could go back to using it.

**Comment by jwnrt** [comment]
@jesultra this PR changes the toolchain (still GNU ARM) from the one provided by CRT to the one provided by https://github.com/hexdae/toolchains_arm_gnu which I believe is more up to date and generates a different binary.

I've tested the binary on our HyperDebug in the office, but wanted to get your opinion on changing the toolchain here.

**Review by pamaury — APPROVED** [review]
Overall this looks to me though I am no blzmod-expert.

**Comment by jesultra** [comment]
> @jesultra this PR changes the toolchain (still GNU ARM) from the one provided by CRT to the one provided by https://github.com/hexdae/toolchains_arm_gnu which I believe is more up to date and generates a different binary.

This project already uses a different toolchain than the one found in ChromeOS "chroot" build environment, meaning that when I do local modifications to the source in platform/ec, and build without using this repository, the resulting binary may behave differently than the official one through this release process.  One time recently, timing or optimization differences let to a bug not being detected when I compiled locally.

What I am saying is that I do not mind at all the toolchain being changed, and furthermore, if in the process we could make it easier to run the same build process locally, and even with modifications to the source code, then that would be a welcome bonus.

File: third_party/chromium/BUILD.bazel
Comment (jesultra) [inline third_party/chromium/BUILD.bazel:16]: The EC Makefile has slightly changed, such that the argument now is called "CROSS_COMPILE" (without the "_arm" suffix).  This change triggers a merge conflict in the other file where this PR moves the code from.

**Review by jesultra — APPROVED** [review]
It seems that after upgrading to Bazel 8, I am no longer able to compile pre-built binaries by running an "Action" on this repository.  I believe this PR is key to a solution.

@jwnrt Please let me know if I can assist in any way.  I have a HyperDebug firmware bugfix, that I would like to publish.

File: third_party/chromium/BUILD.bazel
Comment (jwnrt) [reply third_party/chromium/BUILD.bazel]: Thanks, fixed

**Comment by jwnrt** [comment]
I've rebased to fix that conflict and it seems to build fine. I don't have much time to test it right now but I'm happy for you to merge this if it's working.

> It seems that after upgrading to Bazel 8, I am no longer able to compile pre-built binaries by running an "Action" on this repository. I believe this PR is key to a solution.

Ah, Bazel 8 no longer supports the `WORKSPACE` this repo currently uses. We can try get the action to go back to Bazel 7 or add the `--enable_workspace` flag if this PR doesn't work out.

**Comment by jesultra** [comment]
With this PR, I am able to run `bazel build :hyperdebug-firmware` locally, and the generated `ec.bin` runs on my HyperDebug, so I have high confidence that once this has merged, the GitHub action will also work.

**Pull request merged**
